metadata/moa/scripts/nbconverted/3.align-moa.py

```diff
@@ -0,0 +1,531 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Align Platemap and MOA Data
+# 
+# Here, I align Broad IDs in the Platemap Data to the Most Recent MOA/Target Info
+
+# In[1]:
+
+
+get_ipython().run_line_magic('load_ext', 'nb_black')
+get_ipython().run_line_magic('matplotlib', 'inline')
+
+
+# In[2]:
+
+
+import pathlib
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+
+
+# In[3]:
+
+
+def split_pipe_broad_id(map_subset_df, broad_id_col):
+    # Split a dataframe with broad_id_cols that have pipe delimiters
+    broad_split_df = (
+        pd.DataFrame(
+            map_subset_df.loc[:, broad_id_col]
+            .squeeze()
+            .dropna()
+            .str.split("|")
+            .tolist(),
+            index=map_df.dropna(subset=[broad_id_col]).index,
+        )
+        .stack()
+        .reset_index()
+    )
+    broad_split_df.columns = ["map_index", "_", "broad_id_unique"]
+
+    map_long_df = (
+        map_subset_df.merge(
+            broad_split_df.loc[:, ["map_index", "broad_id_unique"]],
+            left_index=True,
+            right_on="map_index",
+            how="left",
+        )
+        .reset_index(drop=True)
+        .loc[:, ["InChIKey14", "broad_id_unique"]]
+        .drop_duplicates()
+        .assign(broad_date=broad_id_col)
+        .dropna()
+        .reset_index(drop=True)
+    )
+
+    return map_long_df
+
+
+# ## Step 1: Define InChIKey14-to-Broad ID Map
+
+# In[4]:
+
+
+mapping_col = "InChIKey14"
+
+repurposing_col = "broad_id_20170327"
+broad_id_other_dates = ["20170327", "20180516", "20180907", "20200324"]
+
+
+# In[5]:
+
+
+# Load platemap data
+file = pathlib.PurePath("../platemaps/broad_sample_info.tsv")
+all_platemap_df = pd.read_csv(file, sep="\t")
+
+print(all_platemap_df.shape)
+all_platemap_df.head()
+
+
+# In[6]:
+
+
+# How many unique broad IDs?
+len(all_platemap_df.broad_id.unique())
+
+
+# In[7]:
+
+
+# Load MOA map
+map_file = pathlib.PurePath("clue/broad_id_map.csv")
+map_df = pd.read_csv(map_file)
+
+print(map_df.shape)
+map_df.head()
+
+
+# In[8]:
+
+
+# Create MOA map long mapping every Broad ID to InChIKey14
+all_broad_map = []
+for date in broad_id_other_dates:
+    broad_id_col = f"broad_id_{date}"
```

Comment:
IIUC, the `broad_id_{date}` column is quickly dropped in a couple of steps and is not used in any operation in the rest of the code. Is it necessary to create this field or am I missing something?

Reply:
i use `broad_id_{date}` in `split_pipe_broad_id()`. This function isolates the specific `broad_id` and `deprecated_broad_id` column by _date_, splits on `|` and converts the dataframe from wide to long retaining exactly unique `InChIKey14` and `broad_id` combinations.

The `for date in broad_id_other_dates` does this for every date in the InChIKey map

Reply:
I see... I follow the logic now.